Resources/config/orm.xml

```diff
@@ -172,6 +172,14 @@
             <argument type="service" id="doctrine" />
         </service>
 
+        <service id="doctrine.orm.entity_value_resolver" class="Symfony\Bridge\Doctrine\ArgumentResolver\EntityValueResolver">
+            <argument type="service" id="doctrine" />
+            <argument type="service" id="doctrine.orm.entity_value_resolver.expression_language" on-invalid="ignore" />
+            <tag name="controller.argument_value_resolver" priority="-40" />
```

Comment:
Shouldn't this have a higher priority than the `RequestAttributeValueResolver`, I've tested it with the following code & got an error:

```php
    #[Route('/foo/{foo<\d+>}', name: 'app_foo')]
    public function index(Foo $foo): Response
    {
        dd($foo);
    }
```

```
App\Controller\FooController::index(): Argument #1 ($foo) must be of type App\Entity\Foo, string given, called in /vendor/symfony/http-kernel/HttpKernel.php on line 161
```

Since both the route parameter & the method argument have the same name, the `RequestAttributeValueResolver` resolves it to a string which then passed to the controller.

Reply:
Good catch. I'd rather fix `RequestAttributeValueResolver` to make it skip arguments that it are incompatible with the value of the attribute!

Reply:
Actually that won't work because it's legal for a resolver to give incompatible values when there is a controller argument listener that fixes them.
I've registered the entity resolver with priority 110 (the request one is at 100.)